app/src/test/java/com/willowtree/vocable/utils/FaceTrackingPermissionsTest.kt

```diff
@@ -66,60 +61,155 @@ class FaceTrackingPermissionsTest {
+        assertEquals(1, fakePermissionsDialogShower.showPermissionRationaleDialogCalledCount)
+        assertEquals(0, fakePermissionsDialogShower.showSettingsPermissionDialogCalledCount)
     }
 
     @Test
-    fun `when requestFaceTracking unsuccessful, permission state set to disabled`() = runTest {
-        val sharedPreference = createSharedPrefs(headTrackingEnabled = false)
-        val permissions = createFaceTrackingPermissions(sharedPreference)
+    fun `when rationale should be shown, requestFaceTracking() shows dialog and onPositiveClick launches permissionLauncher`() =
+        runTest {
+            val permissions = createFaceTrackingPermissions(
+                shouldShowRequestPermissionRationale = true
+            )
+            advanceUntilIdle()
+
+            permissions.requestFaceTracking()
+
+            assertEquals(1, fakePermissionsDialogShower.showPermissionRationaleDialogCalledCount)
+            assertEquals(0, fakePermissionsDialogShower.showSettingsPermissionDialogCalledCount)
+            fakePermissionsDialogShower.permissionDialogOnPositiveClick.invoke()
+            assertEquals(1, fakePermissionRegisterForLaunch.launchCount)
+        }
+
+    @Test
+    fun `when permission rationale should be shown, rationale dialog shown and onNegativeClick launches permissionLauncher`() =
+        runTest {
+            val sharedPreferences = createSharedPrefs(headTrackingEnabled = true)
+
+            // requestFaceTracking() called implicitly in init block when headTrackingEnabled is true
+            createFaceTrackingPermissions(
+                sharedPreferences = sharedPreferences,
+                shouldShowRequestPermissionRationale = true
+            )
+            advanceUntilIdle()
+
+            assertEquals(1, fakePermissionsDialogShower.showPermissionRationaleDialogCalledCount)
+            assertEquals(0, fakePermissionsDialogShower.showSettingsPermissionDialogCalledCount)
+            fakePermissionsDialogShower.permissionDialogOnNegativeClick.invoke()
+            assertFalse(
+                "Head tracking should be disabled if user denies dialog",
+                sharedPreferences.getHeadTrackingEnabled()
+            )
+        }
+
+    @Test
+    fun `given initial rationale dialog denied, settings dialog rationale shown and updates head tracking when enabled`() = runTest {
+        val sharedPreferences = createSharedPrefs(headTrackingEnabled = false)
+        val permissions = createFaceTrackingPermissions(
+            sharedPreferences = sharedPreferences,
+            shouldShowRequestPermissionRationale = true,
+        )
         advanceUntilIdle()
 
+        // Launch initial rationale dialog, deny permissions
         permissions.requestFaceTracking()
-        assertEquals(permissions.permissionState.first(), IFaceTrackingPermissions.PermissionState.PermissionRequested)
+        fakePermissionsDialogShower.permissionDialogOnPositiveClick.invoke()
```

Comment:
Shouldn't this be `onNegativeClick` if we're denying?

Reply:
ah that's a failure of naming. "positive" click here means the user clicks okay to the rationale dialog so that the system permission dialog is shown. we deny via `fakePermissionRegisterForLaunch.triggerActivityResult`

<img width="319" alt="Screenshot 2024-01-18 at 9 47 15 AM" src="https://github.com/willowtreeapps/vocable-android/assets/58408581/13f615b0-567e-4925-8dc9-b21692393652">

I can make this more clear

Reply:
I've updated `fakePermissionsDialogShower.permissionDialogOnPositiveClick` to `fakePermissionsRationaleDialogShower.rationaleDialogOnPositiveClick` to clarify this